app/scripts/modules/core/src/search/infrastructure/infrastructureNew.controller.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+import _ from 'lodash';
+
+const angular = require('angular');
+
+import { UrlParser } from 'core/navigation';
+import { SearchFilterTypeRegistry } from '../widgets/SearchFilterTypeRegistry';
+import {CLUSTER_FILTER_SERVICE} from 'core/cluster/filter/clusterFilter.service';
+import {CACHE_INITIALIZER_SERVICE} from 'core/cache/cacheInitializer.service';
+import {OVERRIDE_REGISTRY} from 'core/overrideRegistry/override.registry';
+import {RECENT_HISTORY_SERVICE} from 'core/history/recentHistory.service';
+import {PAGE_TITLE_SERVICE} from 'core/pageTitle/pageTitle.service';
+import {INFRASTRUCTURE_SEARCH_SERVICE} from './infrastructureSearch.service';
+import { SearchStatus } from '../searchResult/SearchResults';
+
+module.exports = angular.module('spinnaker.search.infrastructureNew.controller', [
+  INFRASTRUCTURE_SEARCH_SERVICE,
+  RECENT_HISTORY_SERVICE,
+  require('../searchResult/searchResult.directive.js'),
+  PAGE_TITLE_SERVICE,
+  require('./project/infrastructureProject.directive.js'),
+  require('../searchRank.filter.js'),
+  CLUSTER_FILTER_SERVICE,
+  CACHE_INITIALIZER_SERVICE,
+  OVERRIDE_REGISTRY,
+])
+  .controller('InfrastructureNewCtrl', function ($timeout, $scope, infrastructureSearchService, $location,
+                                                 searchService, cacheInitializer, overrideRegistry, pageTitleService,
+                                                 recentHistoryService, $uibModal, $state, clusterFilterService) {
+
+    const ctrl = this;
+    const search = infrastructureSearchService.getSearcher();
+
+    $scope.categories = [];
+    $scope.projects = [];
+
+    ctrl.viewState = {
+      status: SearchStatus.INITIAL,
+      minCharactersToSearch: 3,
+    };
+
+    this.handleFilterChange = function (filters) {
+
+      const params = {};
+      filters.slice(0).reverse().forEach(filter => params[SearchFilterTypeRegistry.getFilterByModifier(filter.modifier).key] = filter.text);
```

Comment:
If the user specifies more than one filter of the same kind, we want the first one to win.  See _issues remaining_ to decide on whether or not we want to allow multiple at all...

Reply:
If we only allow a single filter by kind, we shouldn't allow users to select more than one of each kind - it's not intuitive that other selections will be ignored...